File: ClientApp/src/app/models/auth/UserRegistrationModel.ts
import { BaseUserModel } from "./BaseUserModel";
import { TaskItem } from "../TaskItem";

export class UserRegistrationModel extends BaseUserModel {
    public userName: string

    constructor(email: string, password: string, userName: string) {

        super(email, password);
        this.userName = userName;
    }
}

File: WebApi/Models/Account/UserRegistrationModel.cs
using System.ComponentModel.DataAnnotations;

namespace WebApi.Models.Account
{
    public class UserRegistrationModel 
    {
        [Required]
        public string Email { get; set; }

        [Required]
        public string Password { get; set; }

        public string UserName { get; set; }
    }
}
